### PR TITLE
Update aarch64 building pool to aiinfra-linux-ARM64-CPU-2019

### PR DESCRIPTION
**Pull request by jchen351**
**Description**: Describe your changes.
Update aarch64 building pool to aiinfra-linux-ARM64-CPU-2019
**Motivation and Context**
 This reduce our build time from 3 hours to 16 mins. 
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/c-api-linux-cpu.yml [hunk 1]
```diff
@@ -74,11 +93,10 @@ jobs:
           arch: '${{parameters.OnnxruntimeNodejsBindingArch}}'
           os: 'linux'
           artifactName: 'drop-onnxruntime-nodejs-linux-${{parameters.OnnxruntimeArch}}'
-
-    - template: component-governance-component-detection-steps.yml
-      parameters :
-        condition : 'succeeded'
-
+    - ${{ if not(eq(parameters.OnnxruntimeNodejsBindingArch, 'arm64')) }}:
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/c-api-linux-cpu.yml:96]: Is it enough? By default they would inject such a task to every our build job.

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/templates/c-api-linux-cpu.yml:96]: I also set the 
  variables:
  - name: skipComponentGovernanceDetection
    value: ${{eq('${{parameters.OnnxruntimeNodejsBindingArch}}', 'arm64')}}

File: tools/ci_build/github/azure-pipelines/templates/c-api-linux-cpu.yml [hunk 2]
```diff
@@ -19,14 +19,33 @@ parameters:
...
   steps:
+    - bash: |
+        set -euo pipefail
+        if ! which docker; then
+          sudo apt-get install apt-transport-https ca-certificates curl software-properties-common
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+          sudo add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+          sudo apt-get update
+          sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+          sudo chmod 666 /var/run/docker.sock
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/c-api-linux-cpu.yml:46]: This chmod is a dirty hack, but currently I can't think of a better solution.

**Comment by snnn** [comment]
After this PR gets merged, we can try to revert #11732 .

**Review by snnn — APPROVED** [review]
The "App Center" failure is not related to your change.

**Pull request merged**